core/sql/generator/GenPreCode.cpp

```diff
@@ -170,12 +174,16 @@ static NABoolean processConstHBaseKeys(Generator * generator,
            skey->getIndexDesc()->getOrderOfKeyValues(),
            relExpr->getGroupAttr()->getCharacteristicInputs(),
            TRUE, /* forward scan */
-           exePreds,
+           keyPreds,
            nonKeyColumnSet,
            idesc,
            relExpr->getGroupAttr()->getCharacteristicOutputs(),
            mySearchKeys);
 
+      // Include any remaining key predicates that have not been 
+      // picked up (to be used as the HBase search keys).
+      exePreds += keyPreds;
```

Comment:
Not sure I understand the point of subtracting the keyPreds at line 169 then just adding them back here. The exePreds set is not passed to HbaseSearchKey::makeHBaseSearchKeys.